schema/draft/schema.ts

```diff
@@ -1382,30 +1407,110 @@ export interface RootsListChangedNotification extends Notification {
   method: "notifications/roots/list_changed";
 }
 
+/**
+ * @category elicitation/create
+ */
+export interface ElicitRequestFormParams extends ElicitRequestParams {
+  /**
+   * The elicitation mode.
+   */
+  mode: "form";
+
+  /**
+   * A restricted subset of JSON Schema.
+   * Only top-level properties are allowed, without nesting.
+   */
+  requestedSchema: {
+    type: "object";
+    properties: {
+      [key: string]: PrimitiveSchemaDefinition;
+    };
+    required?: string[];
+  };
+}
+
+/**
+ * @category elicitation/create
+ */
+export interface ElicitRequestURLParams extends ElicitRequestParams {
+  /**
+   * The elicitation mode.
+   */
+  mode: "url";
+
+  /**
+   * The ID of the elicitation, which must be unique within the context of the server.
+   * The client MUST treat this ID as an opaque value.
+   */
+  elicitationId: string;
+
+  /**
+   * The URL that the user should navigate to.
+   *
+   * @format uri
+   */
+  url: string;
+}
```

Comment:
It would be great to add an optional `service` string here to be able to surface the service provider of the upstream resource seeking authentication. 

```diff
{
   "mode": "url",
   "elicitationId": "550e8400-e29b-41d4-a716-446655440000",
   "url": "https://mcp.example.com/ui/oauth/github",
   "message": "Authorize access to your private GitHub repos",
+  "service": "github.com",          // OPTIONAL human-readable hint

}
```

Reply:
That seems tricky in that an MCP Client might trust that value and lead the User to trust it as well, when in reality the `service` value is entirely untrusted and unsigned.

Reply:
Fair point, though the same is true for `message`. It's probably sufficient to rely on the url and message to give enough context to the user on what is being requested and by who.